### ISSUE TITLE
Gobblin Summary Metrics

### ISSUE THREAD

**Issue by apurva-sharma** [linkedin/gobblin#1430]
I enabled metrics on a gobblin job and see it producing the application level metrics:
/path/to/metrics/app_CliMRJob-xxxx/app_CliMRJob-xxxx.txt.metrics.log
As well as the per mapper metrics.
This is a lot of information which is good for application monitoring, but is there a way to get a summary for the metrics?
What if someone is just interested in the total number of records that were published (perhaps by directory).
Is there a way to go about just getting that?

**Comment by ibuenros** [comment]
No, there isn't. There are plans to allow for that, but it probably won't
happen any time soon.

**Comment by apurva-sharma** [comment]
Thanks for the reply, I will try and replicate this functionality through custom HadoopCounters as a work around. Closing this.

**Comment by wosiu** [comment]
Hmm, actually I'm working on something like that. What I have done so far is to extend ScheduledReporter
and override:
1. report() methods in way the do not log metrics by default all the time
2. stopImpl() method, which is called once at the end, and then I call report, which at this point has finall agregated metrics only.

Also I'm going to increased those 2 in properties to something infinit to not call report unnecessarily:
metrics.report.interval=3000000
reporting.interval=50m

And then also we need to implement MyReporterFactory reporter factory which implements CustomReporterFactory
And in properties I just add:
metrics.reporting.custom.builders=gobblin.metrics.MyReporterFactory

And it logs at then end HOWEVER there are no "Meters" metrics which I'm interested most. There are "Times" though. I traversed whole RootMetricContext.get() - still the same.

Moreover if you use custom reporters implemented by LinkedIn (e.g. gobblin.metrics.ConsoleReporterFactory) it doesn't print "Meters" as well which sounds like a bug, but not in Reporter by itstelf.

I wrote more about it here:
https://groups.google.com/forum/#!topic/gobblin-users/3UZkLiWG00Q

I would appreciate any help with my struggle with those missing "Meters" metrics.
